File: mobile/src/main/kotlin/de/scout/fireplace/search/SearchService.java
package de.scout.fireplace.search;

import de.scout.fireplace.models.Search;
import io.reactivex.Single;
import java.util.Map;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.QueryMap;

public interface SearchService {

  @GET("/home/search")
  Single<Search> search(@Query("searchType") String type, @QueryMap Map<String, String> parameters);
}

File: mobile/src/main/kotlin/de/scout/fireplace/search/SearchService.kt
package de.scout.fireplace.search

import de.scout.fireplace.models.Search
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.QueryMap

interface SearchService {

  @GET("/search")
  fun search(@Query("searchType") type: String, @QueryMap parameters: Map<String, String>): Single<Search>
}
